src/dataframe/dataframe.jl

```diff
@@ -1153,13 +1166,49 @@ function Base.append!(df1::DataFrame, df2::AbstractDataFrame; cols::Symbol=:sete
     try
         for (j, n) in enumerate(_names(df1))
             current_col += 1
-            append!(df1[!, j], df2[!, n])
+            if n in _names(df2)
+                df1_c = df1[!, j]
+                df2_c = df2[!, n]
+                S = eltype(df2_c)
```

Comment:
Maybe call this `U` and define it after `T` to use the same order as arguments?

Reply:
I reordered the lines. I would leave `S` and `T` as this is consistent with the rest of the code.